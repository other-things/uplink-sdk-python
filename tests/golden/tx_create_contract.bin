00000000: 00 00 00 00 00 A6 0A 67  6C 6F 62 61 6C 20 69 6E  .......global in
00000010: 74 20 78 20 3D 20 30 20  3B 0A 0A 74 72 61 6E 73  t x = 0 ;..trans
00000020: 69 74 69 6F 6E 20 69 6E  69 74 69 61 6C 20 2D 3E  ition initial ->
00000030: 20 73 65 74 3B 0A 74 72  61 6E 73 69 74 69 6F 6E   set;.transition
00000040: 20 73 65 74 20 2D 3E 20  74 65 72 6D 69 6E 61 6C   set -> terminal
00000050: 3B 0A 0A 40 73 65 74 0A  65 6E 64 20 28 29 20 7B  ;..@set.end () {
00000060: 0A 20 20 74 65 72 6D 69  6E 61 74 65 28 29 3B 0A  .  terminate();.
00000070: 7D 0A 0A 40 69 6E 69 74  69 61 6C 0A 73 65 74 58  }..@initial.setX
00000080: 20 28 69 6E 74 20 79 29  20 7B 0A 20 20 78 20 3D   (int y) {.  x =
00000090: 20 34 32 3B 0A 20 20 74  72 61 6E 73 69 74 69 6F   42;.  transitio
000000A0: 6E 54 6F 28 3A 73 65 74  29 3B 0A 7D              nTo(:set);.}