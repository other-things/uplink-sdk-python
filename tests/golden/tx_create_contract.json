{
    "header": {
        "contents": {
            "contents": {
                "contract": "\nglobal int x = 0 ;\n\ntransition initial -> set;\ntransition set -> terminal;\n\n@set\nend () {\n  terminate();\n}\n\n@initial\nsetX (int y) {\n  x = 42;\n  transitionTo(:set);\n}"
            }, 
            "tag": "CreateContract"
        }, 
        "tag": "TxContract"
    }, 
    "origin": "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve", 
    "signature": "AE4xMTUxMzY4MDA4MjA0NTY4MzM3Mzc5OTQxMjY3NzEzODYwMTUwMjYyODcwOTUwMzQ2MjU2MjM2NDQxODYyNzgxMDg5MjY2OTA3Nzk1Njc6AE04OTQ2OTYxNzM1ODY3Njg2MzUyMjgyODcyNTMxOTE0NTA1MjU1NDk0MDA1MDk0MzU0Nzc5MzIzNTU5NTc4NTcyNTE5NjkxODI4OTQ5Mg=="
}